util/git/client.go

```diff
@@ -741,13 +786,77 @@ func (m *nativeGitClient) RevisionMetadata(revision string) (*RevisionMetadata,
 	authorDateUnixTimestamp, _ := strconv.ParseInt(segments[1], 10, 64)
 	message := strings.TrimSpace(segments[2])
 
+	cmd := exec.Command("git", "interpret-trailers", "--parse")
+	cmd.Stdin = strings.NewReader(message)
+	out, err = m.runCmdOutput(cmd, runOpts{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to interpret trailers: %w", err)
+	}
+	relatedCommits := getReferences(out)
+
 	out, err = m.runCmd("tag", "--points-at", revision)
 	if err != nil {
 		return nil, err
 	}
 	tags := strings.Fields(out)
 
-	return &RevisionMetadata{author, time.Unix(authorDateUnixTimestamp, 0), tags, message}, nil
+	return &RevisionMetadata{
+		Author:     author,
+		Date:       time.Unix(authorDateUnixTimestamp, 0),
+		Tags:       tags,
+		Message:    message,
+		References: relatedCommits,
+	}, nil
+}
+
+// getReferences extracts related commit metadata from the commit message trailers. If related commit
+// metadata is present, we return a slice containing a single metadata object. If no related commit metadata is found,
+// we return a nil slice.
+func getReferences(commitMessageBody string) []RevisionReference {
+	var relatedCommit CommitMetadata
+	for _, line := range strings.Split(commitMessageBody, "\n") {
+		if !strings.HasPrefix(line, "Argocd-reference-commit-") {
+			continue
+		}
+		parts := strings.SplitN(line, ": ", 2)
+		if len(parts) != 2 {
+			continue
+		}
+		trailerKey := parts[0]
+		trailerValue := parts[1]
+		switch trailerKey {
+		case "Argocd-reference-commit-repourl":
+			_, err := url.Parse(trailerValue)
+			if err != nil {
+				log.Errorf("failed to parse repo URL %s: %v", trailerValue, err)
```

Comment:
any ways to have a logger with some context like the revision / and git repo ?

It will be hard to know what caused that error